docker/dev/Dockerfile

```diff
@@ -44,14 +45,21 @@ RUN cd /usr/src/googletest/ && \
      make install && \
      cd / && \
      rm -rf /usr/src/googletest/build
-# Remove unnecessary packages.
-RUN apt-get purge --yes --auto-remove wget \
-  # Erase downloaded archive files.
-  && apt-get clean \
-  && rm -rf /var/lib/apt/lists/ \
-  && set +x
 
-COPY codecompass-build.sh /usr/local/bin
+COPY scripts/install_odb_2.5.sh /tmp/install_odb.sh
+
+RUN chmod u+x /tmp/install_odb.sh && \
+    /tmp/install_odb.sh
+
+
+# Erase downloaded archive files, and unneded packages.
+RUN set +x && \
```

Comment:
Since docker 17 (along with the multistage builds), the creation of multiple layers, does not increase the size of the container as significantly as it did before.
Also it would be more profitable for us if our image consisted of more layers, if we were aiming to use dockerhub in the future, because each layer gets pulled simultaneously, speeding up the process to the maximum capacity of one's computer/internet.

I'm going to omit the package removal part, then :smile:

Reply:
@csordasmarton 
I did some reading on the topic and docker has a feature, to squash all layers into one,  by only keeping the differences, like git does. If we wish to minimize the image size and also keep the advantages of multiple layers while developing, with `docker build --squash` we can.
Because that way we could delete the unneeded packages and data in one, last additional layer, making the final image size even smaller. (This could help to migrate a built image to Dockerhub maybe)
And docker still caches the intermediate layers this way, on the host machine so rebuilding the image does not loose it's incremental features.